docs/dev/libs/cep.md

```diff
@@ -246,63 +333,118 @@ pattern.where(event => ... /* some condition */).or(event => ... /* or condition
 
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
-val nonStrictNext : Pattern[Event, _] = start.followedBy("middle")
+val start : Pattern[Event, _] = Pattern.begin("start")
 {% endhighlight %}
 </div>
 </div>
 
-For non-strict contiguity one can specify if only the first succeeding matching event will be matched, or
-all. In the latter case multiple matches will be emitted for the same beginning.
+Next, you can append more states to your pattern by specifying the desired *contiguity conditions* between them. 
+This can be done using: 
+
+1. `next()`, for *strict*, 
+2. `followedBy()`, for *relaxed*, and 
+3. `followedByAny()`, for *non-deterministic relaxed* contiguity.
+
+or 
+
+1. `notNext()`, if you do not want an event type to directly follow another
+2. `notFollowedBy()`, if you do not want an event type to be anywhere between two other event types
+
+
+<span class="label label-danger">Attention</span> A pattern sequence cannot end in `notFollowedBy()`.
+
+<span class="label label-danger">Attention</span> A `NOT` state cannot be preceded by an optional one.
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
-Pattern<Event, ?> nonStrictNext = start.followedByAny("middle");
+
+// strict contiguity
+Pattern<Event, ?> strict = start.next("middle").where(...);
+
+// relaxed contiguity
+Pattern<Event, ?> relaxed = start.followedBy("middle").where(...);
+
+// non-deterministic relaxed contiguity
+Pattern<Event, ?> nonDetermin = start.followedByAny("middle").where(...);
+
+// NOT pattern with strict contiguity
+Pattern<Event, ?> strictNot = start.notNext("not").where(...);
+
+// NOT pattern with relaxed contiguity
+Pattern<Event, ?> relaxedNot = start.notFollowedBy("not").where(...);
+
 {% endhighlight %}
 </div>
 
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
-val nonStrictNext : Pattern[Event, _] = start.followedByAny("middle")
+
+// strict contiguity
+val strict: Pattern[Event, _] = start.next("middle").where(...)
+
+// relaxed contiguity
+val relaxed: Pattern[Event, _] = start.followedBy("middle").where(...)
+
+// non-deterministic relaxed contiguity
+val nonDetermin: Pattern[Event, _] = start.followedByAny("middle").where(...)
+
+// NOT pattern with strict contiguity
+val strictNot: Pattern[Event, _] = start.notNext("not").where(...)
+
+// NOT pattern with relaxed contiguity
+val relaxedNot: Pattern[Event, _] = start.notFollowedBy("not").where(...)
+
 {% endhighlight %}
 </div>
-
 </div>
-It is also possible to define a temporal constraint for the pattern to be valid.
-For example, one can define that a pattern should occur within 10 seconds via the `within` method. 
+
+Bear in mind that relaxed contiguity means that only the first succeeding matching event will be matched, while
+non-deterministic relaxed contiguity, multiple matches will be emitted for the same beginning.
+
+Finally, it is also possible to define a temporal constraint for the pattern to be valid.
```

Comment:
I would emphasize it applies whole pattern sequence. And if multiple within are applied (even for different states/parts) the shortes one is picked up.